### DISCUSSION TITLE
Foxglove 2.0 Stream Remote Bagfiles Workflow

### DISCUSSION THREAD

**Discussion by frapit** [foxglove/community#1006] [Q&A]
Hi,

we had the following well-working construct in our group:

- [bag database](https://github.com/swri-robotics/bag-database) with all our rosbags self-hosted
- Self-hosted foxglove using the container images from 1.X
- [open with](https://github.com/swri-robotics/bag-database/issues/151) functionality of bag database configured to generate a foxglove URL 

This would generate a link like:
`https://mygroup.org/foxglove/?ds=ros1-remote-bagfile&ds.url=https%3A%2F%2Fmygroup.org%2Fbagdb%2Fbags%2Fdownload%3FbagId%3D15`

possibly with an attached layout ID encoded within the URL.

Since self-hosting foxglove is deprecated since Foxglove 2.0, I am looking to do the same but opening the remote bag with the standalone application on Ubuntu.

This raises multiple questions:

- Can I still pass a remote url above to a foxglove-studio URI handler? Like changing the link to `foxglove:\\open?ds=ros1-remote-bagfile&ds.url=https%3A%2F%2Fmygroup.org%2Fbagdb%2Fbags%2Fdownload%3FbagId%3D15`? I have had no success in opening this on a terminal.
- When trying to insert the link directly insinde foxglove-studio, the pattern does not match, as the remote bag must end with something like `.bag`. This restriction does not apply to the bag-database.
- With all the team layout features, is it still possible to add some layout id within the link? 

Thanks a lot!

**Comment by defunctzombie** [comment]
> * Can I still pass a remote url above to a foxglove-studio URI handler? Like changing the link to `foxglove:\\open?ds=ros1-remote-bagfile&ds.url=https%3A%2F%2Fmygroup.org%2Fbagdb%2Fbags%2Fdownload%3FbagId%3D15`? I have had no success in opening this on a terminal.

This should still work. Which version of the desktop app are you using? On which OS? Maybe share your entire CLI command so we can try to reproduce?

You can also build a similar a url for the web app `https://app.foxglove.dev/view?ds=ros1-remote-bagfile&ds.url=https%3A%2F%2Fmygroup.org%2Fbagdb%2Fbags%2Fdownload%3FbagId%3D15`

> * When trying to insert the link directly insinde foxglove-studio, the pattern does not match, as the remote bag must end with something like `.bag`. This restriction does not apply to the bag-database.

This is a requirement of the "open remote file" feature - you need an extension on the URL so the app can know which kind of data source to use to read the file. We could look at doing some early identification of the file (by header or footer bytes) to allow urls with no extension, but this is the first time I am aware this has come up as most folks using this feature have had extensions on their URLs.

> * With all the team layout features, is it still possible to add some layout id within the link?

Yep. You would have a `layoutId` in the URL and point it to one of the team layouts. Whomever opens the URL will open that layout as well.

**Reply by frapit** [reply]
Thanks for these hints! I was using Foxglove 2.6.1 on Ubuntu 24. The issue is resolved -  I missed some `Could not set app as handler for foxglove://` in the logs. I added the handler and now foxglove will open. 
It still refrains from opening the remote bag due to the missing extension. (Error `Could not open connection: Unsupported extension: ` popping up.) The bag database backend creates a bagfile stream but exposes it only as ID in the frontend or download / streaming links, e.g. the links are constructed like that: `https://group.org/bagdb/bags/download?bagId=15` I understand that this might be difficult to handle since it could also be `mcap` file, etc.
